lang/py/avro/io.py

```diff
@@ -936,6 +945,18 @@ def read_record(self, writers_schema, readers_schema, decoder):
       else:
         self.skip_data(field.type, decoder)
 
+    if self.unit_db is not None:
+      wsid = str(id(writers_schema))
```

Comment:
Schema fingerprint might work too - I just wanted something to uniquely identify the schema, that was cheap to compute. I'm not very familiar with the properties of schema fingerprints, except that they seem to come in multiple hash sizes.

Reply:
The motivation for tagging it at all was to save redundant compute - don't resolve things that have already been resolved. It touches on the fact that schema resolution happens for each data element, which itself seems redundant to me, except in the particular case where unions are in play.
https://issues.apache.org/jira/browse/AVRO-2748

Reply:
Right, the memoization is a good idea. I'm just asking if the fingerprint would be an effective key. If it is, then two instances created in running `schema.parse` on the same schema string would have the same wsid, and so the memoization would have a wider net.

Reply:
I'm confident fingerprint would work. It might also be more portable across avro language implementations. My main concern is that computing a hash on a schema is itself expensive, at least compared to "get the memory address of this object".  Are schema fingerprints themselves memoized?

Reply:
Ugh, sorry, i don't think python has a fingerprint implementation at all just yet. There was work on it, but I can't find it in the code now (looking on my phone, anyway). As I recall, it ought to be quite memoizable.